envoy/config/filter/accesslog/v2/accesslog.proto

```diff
@@ -147,6 +147,20 @@ message ResponseFlags {
 
   // Indicates that the request was rate-limited locally.
   bool rate_limited = 12;
+
+  message Unauthorized {
+    // Reasons why the request was unauthorized
+    enum UnauthorizedType {
```

Comment:
I would rename this to `Reason` (it's already qualified as Unauthorized by the outer scope).